### PR TITLE
Add Qwen CPP Pipeline

### PR DESCRIPTION
**Pull request by sammysun0711**
Draft PR to add Qwen CPP pipeline, will swtich to use OV tokenizer for Qwen tokenization pipeline when it available.

### CODE REVIEW

File: llm/qwen_cpp/qwen.tiktoken
Comment (ilya-lavrenov) [inline llm/qwen_cpp/qwen.tiktoken:1]: is it possible to provide a command how to download this file to README.md ? and remove from repo.
Similarly, why do we need ABSEIL dependency?

File: llm/qwen_cpp/unordered_dense.h
Comment (ilya-lavrenov) [inline llm/qwen_cpp/unordered_dense.h:1]: is it possible to use it as submodule as well?
where is the origin of this file?

Comment (ilya-lavrenov) [reply llm/qwen_cpp/unordered_dense.h:1]: patches can be used, example is https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/custom_operations/user_ie_extensions/tokenizer/CMakeLists.txt#L59

Comment (sammysun0711) [reply llm/qwen_cpp/unordered_dense.h:1]: Thanks for your suggestion and example, will checkout.

File: llm/qwen_cpp/main.cpp [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+      ireq.infer();
+      duration_ms = get_duration_ms_until_now(startTime);
+      std::cout << "First inference took " << duration_ms << " ms" << std::endl;
+
+      // Get first inference results
+      size_t vocab_size = ireq.get_tensor("logits").get_shape().back();
+      float* logits = ireq.get_tensor("logits").data<float>() + (input_ids.size() - 1) * vocab_size;
+      out_token = int32_t(std::max_element(logits, logits + vocab_size) - logits);
+      if (text_streamer) {
+        text_streamer->put({out_token});
```
Comment (ilya-lavrenov) [inline llm/qwen_cpp/main.cpp:227]: should not we `put` our `input_ids` as well?
Looks like `text_streamer` expects prompt first, then `put` should be called with token by token

File: llm/qwen_cpp/qwen.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    if (!(cond)) \
+    QWEN_THROW << "check failed (" #cond ") "
+class BaseStreamer{
+  public:
+    virtual ~BaseStreamer() = default;
+    virtual auto put(const std::vector<int> &output_ids) -> void = 0;
+    virtual auto end() -> void = 0;
+};
+
+class StreamerGroup : public BaseStreamer {
```
Comment (ilya-lavrenov) [inline llm/qwen_cpp/qwen.h:40]: looks like `StreamerGroup` is not used

File: llm/qwen_cpp/main.cpp [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    }
+
+    if (args.device.find("GPU") != std::string::npos) {
+        device_config[ov::cache_dir.name()] = "llm-cache";
+        device_config[ov::intel_gpu::hint::queue_throttle.name()] = ov::intel_gpu::hint::ThrottleLevel::MEDIUM;
+        device_config[ov::intel_gpu::hint::queue_priority.name()] = ov::hint::Priority::MEDIUM;
+        device_config[ov::intel_gpu::hint::host_task_priority.name()] = ov::hint::Priority::HIGH;
+        device_config[ov::hint::enable_cpu_pinning.name()] = true;
+        device_config[ov::enable_profiling.name()] = false;
+    }
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply llm/qwen_cpp/main.cpp:147]: @sammysun0711 why do we need these custom settings and cannot rely on OV defaults?

[the remark this answers is not in the document]
Comment (sammysun0711) [reply llm/qwen_cpp/main.cpp:147]: @ilya-lavrenov, yes, we need custom setting for iGPU optimization on hybrid core platform: 
https://github.com/openvinotoolkit/openvino/pull/21261/files, e.g. set enable_cpu_pinning to bind CPU for model compilation on GPU.

File: llm/qwen_cpp/main.cpp (hunk 1 above)
Comment (sammysun0711) [reply llm/qwen_cpp/main.cpp]: Removed

File: llm/qwen_cpp/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    2. https://huggingface.co/Qwen/Qwen-7B-Chat
+    3. https://huggingface.co/Qwen/Qwen-7B-Chat-Int4
+
+## Qwen Demo
+### Step 1: Download and convert the model
+
+```sh
+cd openvino.genai/llm_bench/python
+python -m pip install -r requirements.txt
+python convert.py --model_id Qwen/Qwen-7B-Chat-Int4 --output_dir Qwen-7B-Chat-GPTQ_INT4_FP16-2K --precision FP16
```
Comment (Wovchena) [inline llm/qwen_cpp/README.md:25]: Is there a way to convert without cuda?

For CPU PyTorch I get `AssertionError: Torch not compiled with CUDA enabled`, and for full PyTorch version I get `RuntimeError: Found no NVIDIA driver on your system.`

Comment (sammysun0711) [reply llm/qwen_cpp/README.md:25]: Yes, pls check: https://github.com/openvinotoolkit/openvino.genai/blob/master/llm_bench/python/NOTES.md#qwen-7b-chat-int4---torch-not-com

**Comment by Wovchena** [comment]
`TextStreamer` is going to be added by https://github.com/openvinotoolkit/openvino.genai/pull/130

**Pull request closed without merging**